src/ol/source/Cluster.js

```diff
@@ -132,7 +140,23 @@ class Cluster extends VectorSource {
   }
 
   /**
-   * handle the source changing
+   * Replace the wrapped source.
+   * @param {VectorSource|null} source The new source for this instance.
+   * @api
+   */
+  setSource(source) {
+    if (this.source) {
+      this.source.removeEventListener(EventType.CHANGE, this.boundRefresh_);
+    }
+    this.source = source;
+    if (source) {
+      this.sourceListenKey_ = source.addEventListener(EventType.CHANGE, this.boundRefresh_);
```

Comment:
The way you register the listener, you do not need the `sourceListenKey_` to unregister.

Reply:
I first used on and unByKey to add and remove the listeners and this slipped through.